### DISCUSSION TITLE
VS 2022 Copilot authenticated with wrong account

### DISCUSSION THREAD

**Discussion by nathaleaa** [community/community#13587] [Copilot Conversations] [answered]
I got access to Copilot Technical Preview yesterday, and i installed the extension on Visual Studio Community 2022.
When it prompted me to authenticate and opened github website to paste code, i haven't noticed that i'm logged on wrong account on GitHub. I got an information that i don't have access to Copilot on this account, and i'm stuck with it.

I tried reinstalling the extension, and even entire IDE, but it seems that it still remember the wrong account.
I counldn't find an option to re-authenticate, nor any solution to this on the internet, as most threads are related to vscode, not vs 2022

![image](https://user-images.githubusercontent.com/67127030/160252640-71909678-16db-4803-8204-1f3fbc7e8675.png)
![image](https://user-images.githubusercontent.com/67127030/160252716-b40fcb28-7793-493b-94dd-2d1382a78c61.png)

UPDATE
after a few days the issue is still present. I tried shuffling accounts, reinstalling IDE again after rebooting and clearing temporary files, it seems like neither of these worked.

When i right click copilot i see "sign in" option, but when i click it it automatically tryies to connect with that wrong account
![image](https://user-images.githubusercontent.com/67127030/160678053-c0c35ca8-dd8d-476b-9785-ba0ac54e2d7c.png)

UPDATE 2
OK, in an act of desperacy i reinstalled entire OS, it seems that it worked, but i wouldn't consider it as a solution. I had nothing to lose, beacuse it was a fresh system anyway, but i can't imagine many people willing to factory reset their PC.

**Comment by kzu** [comment, marked as answer]
The issue is not on the VS side, but on the browser side. Visual Studio simply opens your default browser and navigates to the device authentication page on github.com. If you sign-out on the browser from the "wrong" account, you'd get asked to sign-in again on github.com, at which point you'd enter the device code on the "right" account. The extension does not automate the browser in any way other than asking to navigate you to the relevant page.

Hope this helps!

**Reply by nathaleaa** [reply]
Hmm, weird, i've tried that aswell, i even switched my default browser, but it seemed like it had no effect, although it is possible that i had missed something. 
Thanks for your answer!

**Reply by FuSioN79m** [reply]
Trying to exit of all account, trying to delete github account in VS 2022.
Still on "You don't have access to GitHub Copilot." when I click on "Sign In"
It doesn't ask to me credentials! It still remember the wrong account.

Maybe can be a block on network communication but I don't know on what.

**Reply by CraigWarfordAdaptX** [reply]
Finally a reasonable answer.  (As opposed to removing VSCode and reinstalling everythin.g)  I was able to fix the issue using this method.  However, this also shows there is a problem with how the Github authentication is done through the browser, because I have multiple Chrome profiles (work and personal), and it was attempting to authenticate through the wrong profile.
